### PR TITLE
Fix: DO-2086 update edge constraints in visual edge encoder

### PR DESCRIPTION
**Pull request by patricia-causalens**
## Motivation and Context

With release of 0.3.0 of `cai-causal-graph` there have been changes to how constraint edges are defined:
https://github.com/causalens/cai-causal-graph/commit/e47ee4357fc5ec8291c3d548438b4f4fbea1170f

## Implementation Description

Update `EdgeConstraintType` to have the same values as the newly defined. At the moment I have made so that SOFT_DIRECTED and HARD_DIRECTED will behave and look in the same way. We can later add different visual representations for them once we have an idea for what this should be. 

There will be accompanying PRs across dara, and ui internal, dara internal adapting to these changes.

## Any new dependencies Introduced

None

## How Has This Been Tested?

Locally on Storybook

## PR Checklist:

- [x] I have implemented all requirements? (see JIRA, project documentation).
- [x] I am not affecting someone else's work, If I am, they are included as a reviewer.
- [ ] I have added relevant tests (unit, integration or regression).
- [ ] I have added comments to all the bits that are hard to follow.
- [ ] I have added/updated Documentation.
- [x] I have updated the appropriate changelog with a line for my changes.

### CODE REVIEW

File: packages/ui-causal-graph-editor/src/types.tsx [hunk 1]
```diff
@@ -203,11 +203,10 @@ export type SimulationGraph = AbstractGraph<SimulationNode, SimulationEdge, Simu
  * Int values from cl_interfaces.EdgeConstraint
  */
 export enum EdgeConstraintType {
```
Comment (krzysztof-causalens) [inline packages/ui-causal-graph-editor/src/types.tsx:205]: Is the comment above the still accurate or did the definition move? It's somewhere in `cai-causal-graph` I think?

**Review by tamara-causalens — APPROVED** [review]

**Review by krzysztof-causalens — APPROVED** [review]
LGTM. Let's make sure this is released as a minor version as we use the `~` version specification everywhere

File: packages/ui-causal-graph-editor/src/types.tsx (hunk 1 above)
Comment (patricia-causalens) [reply packages/ui-causal-graph-editor/src/types.tsx:205]: good catch, it has both changed places and type, fixed

**Comment by patricia-causalens** [comment]
> LGTM. Let's make sure this is released as a minor version as we use the `~` version specification everywhere

Sounds good, have one more change to be done here before the release, then will make a minor

**Pull request merged**